test/ArborXTest_TreeTypeTraits.hpp

```diff
@@ -0,0 +1,83 @@
+/****************************************************************************
+ * Copyright (c) 2012-2020 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_TEST_TREE_TYPE_TRAITS_HPP
+#define ARBORX_TEST_TREE_TYPE_TRAITS_HPP
+
+#include <ArborX.hpp>
+
+#include <tuple>
+
+// NOTE Because std::tuple does not take template template parameters
+template <template <class> class...>
+struct Tuple
+{
+};
+
+#ifndef ARBORX_TEST_TREE_TYPES
+template <class MemorySpace>
+using ArborX__BoundingVolumeHierarchy =
+    ArborX::BoundingVolumeHierarchy<MemorySpace>;
+#define ARBORX_TEST_TREE_TYPES Tuple<ArborX__BoundingVolumeHierarchy>
```

Comment:
I'm not sure where to put it, but it would be helpful to have a comment somewhere explaining existence of double underscores and such.